libs/pages/application/src/lib/feature/page-settings-deployment-restrictions-feature/page-settings-deployment-restrictions-feature.tsx

```diff
@@ -0,0 +1,156 @@
+import { useParams } from 'react-router-dom'
+import { useFetchEnvironmentDeploymentRule } from '@qovery/domains/environment'
+import { useDeleteDeploymentRestriction, useDeploymentRestrictions } from '@qovery/domains/services/feature'
+import {
+  BannerBox,
+  BlockContent,
+  Button,
+  ButtonIcon,
+  ButtonIconStyle,
+  EmptyState,
+  HelpSection,
+  IconAwesomeEnum,
+  InputText,
+  useModal,
+  useModalConfirmation,
+} from '@qovery/shared/ui'
+import CrudModalFeature from './crud-modal-feature/crud-modal-feature'
+
+export function PageSettingsDeploymentRestrictionsFeature() {
+  const { projectId = '', environmentId = '', applicationId = '' } = useParams()
+  const serviceParams = {
+    serviceId: applicationId,
+    serviceType: 'APPLICATION' as const,
+  }
+  const { data: deploymentRestrictions = [] } = useDeploymentRestrictions(serviceParams)
+  const { mutate: deleteRestriction } = useDeleteDeploymentRestriction(serviceParams)
+  const { openModal, closeModal } = useModal()
+  const { openModalConfirmation } = useModalConfirmation()
+  const { data: environmentDeploymentRules } = useFetchEnvironmentDeploymentRule(projectId, environmentId)
+  const isAutoDeployActive = environmentDeploymentRules?.auto_deploy
+
+  const handleCreate = () => {
+    openModal({
+      content: <CrudModalFeature onClose={closeModal} {...serviceParams} />,
+    })
+  }
+
+  const handleEdit = (deploymentRestriction: ApplicationDeploymentRestriction) => {
+    openModal({
+      content: (
+        <CrudModalFeature onClose={closeModal} deploymentRestriction={deploymentRestriction} {...serviceParams} />
+      ),
+    })
+  }
+
+  const handleDelete = (deploymentRestriction: ApplicationDeploymentRestriction) => {
+    openModalConfirmation({
+      title: 'Delete Restriction',
+      name: `${deploymentRestriction.mode}/${deploymentRestriction.type}/${deploymentRestriction.value}`,
+      isDelete: true,
+      action() {
+        deleteRestriction({
+          serviceId: applicationId,
+          serviceType: 'APPLICATION',
+          deploymentRestrictionId: deploymentRestriction.id,
+        })
+      },
+    })
+  }
+
+  return (
+    <div className="flex flex-col justify-between w-full">
+      <div className="p-8 max-w-content-with-navigation-left">
+        <div className="flex justify-between mb-8">
+          <div>
+            <div className="flex justify-between mb-2 items-center">
+              <h3 className="text-neutral-400 text-lg">Deployment Restrictions</h3>
+            </div>
+
+            <p className="text-sm text-neutral-400 max-w-lg">
+              Specify which changes in your repository should trigger or not an auto-deploy of your application
+            </p>
+          </div>
+          <Button dataTestId="add-button" onClick={handleCreate} iconRight={IconAwesomeEnum.CIRCLE_PLUS}>
+            New Restriction
+          </Button>
+        </div>
+
```

Comment:
Can you add a spinner with `<LoaderSpinner />` if we didn't have the data ? You can find it on the Domains setting section. It should remove the glitch if you hard reload the page

https://www.loom.com/share/e9bfd224a33f46c2969a823f12dafe55

Reply:
fixed